agent/consul/autopilot_test.go

```diff
@@ -449,26 +449,30 @@ func TestAutopilot_BootstrapExpect(t *testing.T) {
 			if mem.IsLeader() && leader {
 				return mem
 			}
-			if !leader && !mem.IsLeader() {
+			if !mem.IsLeader() && !leader {
```

Comment:
We definitely don't need to do it an we might argue that it is less clear but this should work to be able to replace both ifs

```
if mem.IsLeader() == leader {
   return mem
}
```

Your way is almost certainly clearer but I couldn't help but think this up.